lib/discover/manager.go

```diff
@@ -34,8 +34,11 @@ import (
 type Manager interface {
 	FinderService
 	ChildErrors() map[string]error
+	DiscoveryStatus() map[string]DiscoveryStatusEntry
 }
 
+type DiscoveryStatusEntry *string
```

Comment:
What's the purpose of this instead of an error?

I mean I know about the json serialization shenanigans on the api side, but that doesn't (shouldn't) impact the choices here.

Reply:
There's already a `errorString` convenience function to convert `error` to something json-friendly. I don't think it's sensible to avoid errors in go apis just to avoid that. And for future extensibility (e.g. adding fields), shouldn't it be `type DiscoveryStatusEntry struct{ err error }` to end up in a json object?

Reply:
There is even an `errorStringMap()` which seems to do just the right thing.  Will look into it and then probably revert everything back to just using `ChildErrors()` again.

I didn't switch to a struct right away to avoid a breaking change in the JSON API.  I get the feeling you're not reading my commit messages in full, which usually explain such things, e.g. https://github.com/syncthing/syncthing/pull/7539/commits/c793e08b1229ebf604ea8d78fea7440abab9aedf and https://github.com/syncthing/syncthing/pull/7539/commits/177058353f55d5e311265085810b0d881389b2e0 in this case.  Just a hint so there might be less questions on the next review :-)